pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -133,8 +135,10 @@ func (m *mergeQueryable) Querier(mint int64, maxt int64) (storage.Querier, error
 		idLabelName:        m.idLabelName,
 		callbacks:          m.callbacks,
 		upstream:           upstream,
+		resolver:           tenant.NewMultiResolver(),
```

Comment:
There's no reason to rely on the global state of the default resolver in the `tenant` module.